src/jit/lsra.cpp

```diff
@@ -4897,7 +4924,9 @@ LinearScan::allocateBusyReg(Interval *current, RefPosition *refPosition)
         if (assignedInterval == nullptr)
         {
             RefPosition* nextPhysRegPosition = physRegRecord->getNextRefPosition();
-            assert(nextPhysRegPosition->nodeLocation == refLocation && candidateBit != candidates);
+
+            // nodeLocation is not always == refLocation, so this assert is removed
+            //assert(nextPhysRegPosition->nodeLocation == refLocation && candidateBit != candidates);
             continue;
```

Comment:
The issue here is that the call to tryAllocateReg fails, so the register allocator is asserting that all of the candidates must either be 1) already assigned to a register, or 2) have a fixed reference at this location that is NOT this reference. Note that this should really be treated as if it is a fixed reference of BOTH registers. Probably a better workaround for now would be to only set a single bit in the dstCandidates, and then to add an internal register with the other bit set in its candidates. That way both will be considered fixed registers, and you won't have the current issue, where another interval is assigned to the register because it doesn't know it is going to be killed.